script/admin.js

```diff
@@ -0,0 +1,165 @@
+                    $renameForm.submit();
+                }
+            }
+        },
+        {
+            text: LANG.plugins.tagging.admin_cancel,
+            click: function () {
+                actionDialog.dialog('close');
+            }
+        },
+    ];
+
+    /**
+     * Modal for further user interaction
+     *
+     * @type {jQuery}
+     */
+    const actionDialog = jQuery("#tagging__action-dialog").dialog({
+        autoOpen: false,
+        height: 400,
+        width: 300,
+        modal: true,
+        buttons: actionDialogButtons,
+        close: function () {
+            actionDialog.html('');
+        },
+        open: function( event, ui ) {
+            actionDialogHtml();
+        }
+    });
+
+    /**
+     * Modal listing pages with a given tag
+     *
+     * @type {jQuery}
+     */
+    const taggedPagesDialog = jQuery("#tagging__taggedpages-dialog").dialog({
+        autoOpen: false,
+        height: 400,
+        width: 400,
+        modal: true,
+        close: function () {
+            taggedPagesDialog.html('');
+        },
+        open: function( event, ui ) {
+            taggedPagesgHtml();
+        }
+    });
+
+    /**
+     * Injects dialog contents that match the triggered action
+     */
+    const actionDialogHtml = function() {
+
+        const actionData = actionDialog.dialog('option', 'data');
+        let $renameForm;
+
+        if (actionData.action === 'delete') {
+            actionDialog.append('<h1>' + LANG.plugins.tagging.admin_delete + ' ' + actionData.tid + '</h1>');
+            actionDialog.append('<p>' + LANG.plugins.tagging.admin_sure + '</p>');
+        } else if (actionData.action === 'rename') {
+            actionDialog.append('<h1>' + LANG.plugins.tagging.admin_rename + ' ' + actionData.tid + '</h1>');
+            actionDialog.append('<p>' + LANG.plugins.tagging.admin_newtags + ' </p>');
+            actionDialog.append('<form id="tagging__rename"><input type="text" name="newtags" id="tagging__newtags"></form>');
+
+            $renameForm = jQuery('#tagging__rename');
+            $renameForm.on('submit', function( event ) {
+                event.preventDefault();
+                const newValue = jQuery(this).find('#tagging__newtags').val();
+                callBackend({call: 'plugin_tagging_rename', tagging: {oldValue: actionData.tid, newValue: newValue} });
+            });
+        }
+        actionDialog.append('<p class="warning">' + LANG.plugins.tagging.admin_warning_all + '</p>');
+    };
+
+    /**
+     * Displays tagged pages
+     */
+    const taggedPagesgHtml = function() {
+
+        const data = taggedPagesDialog.dialog('option', 'data');
+        const pids = data.pids.split(/,\s*/);
+
+        taggedPagesDialog.append('<h1>Tagged pages</h1>');
+        taggedPagesDialog.append('<ul>');
+        pids.forEach(function (pid) {
+            taggedPagesDialog.append('<li>' + pid + '</li>');
+        });
+        taggedPagesDialog.append('</ul>');
```

Comment:
I don't like that the content of this dialog is build in JavaScript. I'd prefer the full HTML coming from the backend via the AJAX call. That way we can add more functionality in an easier way. First thing to do: each of those page IDs should be linked to open the page.

Reply:
Done